### DISCUSSION TITLE
context reference in flattened json+ld

### DISCUSSION THREAD

**Discussion by srerickson** [piprate/json-gold#75] [Q&A] [answered]
I'm hoping to use `json-gold` to build json-ld like the following:

```json
{
   "@context": "https://w3id.org/ro/crate/1.1/context", 
   "@graph": [
      {
         "@id": "./",
         "identifier": "1234",
         "@type": "Dataset",
      }
   ]
}
```

My code looks like this: 

```go
proc := ld.NewJsonLdProcessor()
options := ld.NewJsonLdOptions("")
newDoc := map[string]any{"@type": "http://schema.org/Dataset", "@id": "./"}
val := map[string]any{"@value": "123"}
ld.AddValue(newDoc, "http://schema.org/identifier", val, true, false, false, false)
ctx := map[string]any{
   "@context": roContext,
}
flat, err := proc.Flatten(newDoc, ctx, options)
if err != nil {
    log.Fatal(err)
}
enc := json.NewEncoder(os.Stdout)
enc.SetIndent("", "  ")
enc.Encode(newDoc)
enc.Encode(flat)
```

I'm noticing that the resulting `@context` is fully resolved. Instead of this: 

```json
 "@context": { ... }
```

I'd like:

```json
"@context": "https://w3id.org/ro/crate/1.1/context", 
```

I can manually change the value, but I was wondering if there's a better way.

Thanks!

**Comment by kazarena** [comment]
@srerickson,

I need to check the definition of the compaction algorithm to confirm the prescribed behaviour. I think you are right and it should preserve the shape of the given context. If so, we'll fix it.

As a side observation, if you pass the context as a URL, the algorithm will be downloading the context definition _for every invocation_ of `Flatten` operation (unless you use context caching). To make it more efficient, it may be a good idea to load the context into memory, pass it to the function, and then substitute the context representation by URL after flattening.

**Reply by srerickson** [reply, marked as answer]
Thanks for the response - that's helpful!
